### DISCUSSION TITLE
20240926 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#109] [Meetings]
- Time:  9:30 in Canary Islands (WET (UTC+0)).
- Time: 10:30 in Vaduz (CET (UTC+1)).

Location:
<https://meeting.blockfinance-eco.li/NautilusCyberneeringWeeklyDevMeeting>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- Cameron Garnham(@da2ce7)  - x 
- Jose Celano(@josecelano) - x
- Constantin Bosse(@cgbosse) - x
- Mario Melchor(@mario-nt) - x
- Graeme Byrne(@grmbyrn) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
### Torrust Solution

Released version `3.0.0-rc.1` (updating dependencies in some cases):

- Tracker: https://github.com/torrust/torrust-tracker/releases/tag/v3.0.0-rc.1
- Index: https://github.com/torrust/torrust-index/releases/tag/v3.0.0-rc.1
- Index GUI: https://www.npmjs.com/package/torrust-index-gui/v/3.0.0-rc.1
  - Types: https://www.npmjs.com/package/torrust-index-types-lib
  - API: https://www.npmjs.com/package/torrust-index-api-lib

### Torrust Website

- Created a new repo for the new version: https://github.com/torrust/torrust-website-v2 (to deploy it without affecting the current one).
- Review version 2: https://v2.torrust.com/ opening [new issues](https://github.com/torrust/torrust-website-v2/issues).
- Working on contents **WIP** (working on Index page).

PR reviews:

- https://github.com/torrust/torrust-website-v2/pull/44
- https://github.com/torrust/torrust-website-v2/pull/23
- https://github.com/torrust/torrust-website-v2/pull/14
- https://github.com/torrust/torrust-website-v2/pull/2

Issues:

- https://github.com/torrust/torrust-website-v2/issues/4

### Torrust Index

Meeting with @mario-nt: helping him fixing issues with E2E tests for authorization service.

### Torrust Index GUI

PR review:

- https://github.com/torrust/torrust-index-gui/pull/625

### Torrust Bencode2Json

- Implemented almost 100% (missing escaping strings: `"`, `\\`, JSON).
- Added more tests (only for valid cases).
- Minor refactorings.

**TODO:**

- More refactorings.
- A few more tests for valid cases.
- Tests for error cases and remove "panics" (return errors with context).
- And more ...

**Comment by mario-nt** [comment]
### Index
	
[https://github.com/torrust/torrust-index/issues/615 - tests](https://github.com/torrust/torrust-index/pull/718/)


- 1 to 1 with Jose
- Start authorization tests from scratch with new approach

### GUI

- https://github.com/torrust/torrust-index-gui/pull/625

### Training

- SVG elements 101
- Flexbox 101
